backend/session/direct-ipc.c

```diff
@@ -129,7 +130,10 @@ static void communicate(int sock) {
 	struct msg msg;
 	int drm_fd = -1;
 	bool running = true;
+	fd_set mastered_fds;
```

Comment:
I'm not a fan of using an fd_set for this, considering the fairly low maximum file descriptor (usually 1024). It's not particularly likely that we'll reach this number, but it's not the kind of limitation I would want to add to library code.

Reply:
Ah, I thought that the limit on fd_set was also the largest possible file descriptor. You're right that it isn't, though, and there is a remark about it in the BUGS section of the fd_set manpage.
I'll change it to a wl_array.